solcore/absorption_calculator/adachi_alpha.py

```diff
@@ -18,27 +18,36 @@ def create_adachi_alpha(material, Esteps=(1.42, 6, 3000), T=300, wl=None):
                       'S.Adachi, “Optical dispersion relations for GaP, GaAs, GaSb, InP, InAs, InSb, AlxGa1−xAs, '
                       'and In1−xGaxAsyP1−y” J.Appl.Phys., vol.66, no.12, pp.6030–12, 1989.')
 
+    # determine if material is stored in Solcore as an alloy by checking main_fraction
+    # if not, can simply look up by name. Otherwise, need to pass relevant alloy compositions also
+
     if type(material) != str:
```

Comment:
This is confusing (and coming from the old times): if the function expects `material` to be a Solcore material as stated in the docstring then it does not make sense to check if it is not a string - it should not be a string!

If it is permitted not to be a Solcore material and be a string instead, then you need to handle that case in order to provide a value for `alloy_args`, in addition to updating the docstring. As it is now, this will fail if `material` is a string because `alloy_args` will not exist.